src/Bridge/Symfony/Bundle/EventListener/EventDispatcher.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\EventListener;
+
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class EventDispatcher
+{
+    private $eventName;
+    private $eventClass;
+    private $dispatcher;
+
+    public function __construct(string $eventName, string $eventClass, EventDispatcherInterface $dispatcher)
+    {
+        $this->eventName = $eventName;
+        $this->eventClass = $eventClass;
+        $this->dispatcher = $dispatcher;
+    }
+
+    public function dispatch(Event $event): void
+    {
+        $internalEvent = null;
+
+        switch ($event) {
+            case $event instanceof GetResponseEvent:
+                $internalEvent = new $this->eventClass(null, ['request' => $event->getRequest()]);
```

Comment:
How about putting the original event in the internal event ?

Reply:
Not sure about this. I want to decouple from Symfony as much as possible. What is the use case?

Reply:
The `GetResponseEvent` from Symfony has a response part which is not in the internal event, it could be useful to have it here.

Reply:
Yes but it's only to set the response and bypass everything else. I'm not sure we want this in API Platform, do we? If there are a "respose" (controller result) and a request (in `GetResponseForControllerResultEvent` for instance), I will add them both to the context.

Reply:
Yup, but then what if we get an exception event (`GetResponseForExceptionEvent` I think ?), and we want to get the thrown exception ? Same for all the other sub controllers.

Reply:
I will add an `exception` field in the context then. IMO we want as much information as the Symfony event but we don't want to use the event itself since it's too tight to Symfony.

Reply:
Then the case should be split accordingly depending on the event it handles (exception, controller result, ... and so on)